src/ssl/support.cc

```diff
@@ -390,9 +390,17 @@ ssl_verify_cb(int ok, X509_STORE_CTX * ctx)
 }
 
 void
-Ssl::SetupVerifyCallback(Security::ContextPointer &ctx)
+Ssl::SetupPeerVerification(Security::ContextPointer &ctx, bool verify, bool fail_if_no_peer_cert)
 {
-    SSL_CTX_set_verify(ctx.get(), SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, ssl_verify_cb);
+  int mode;
+
+  mode = SSL_VERIFY_NONE;
+  if(verify)
+    mode |= SSL_VERIFY_PEER;
```

Comment:
My reading of OpenSSL [docs](https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_verify.html) suggests that SSL_VERIFY_PEER and SSL_VERIFY_FAIL_IF_NO_PEER_CERT should not be added to SSL_VERIFY_NONE, even it it happens to work in your tests:

> If the mode is SSL_VERIFY_NONE none of the other flags may be set.